provisioner/terraform/executor.go

```diff
@@ -0,0 +1,338 @@
+	}
+	resources, err := e.planResources(ctx, planfilePath)
+	if err != nil {
+		return nil, err
+	}
+	return &proto.Provision_Response{
+		Type: &proto.Provision_Response_Complete{
+			Complete: &proto.Provision_Complete{
+				Resources: resources,
+			},
+		},
+	}, nil
+}
+
+func (e executor) planResources(ctx context.Context, planfilePath string) ([]*proto.Resource, error) {
+	plan, err := e.showPlan(ctx, planfilePath)
+	if err != nil {
+		return nil, xerrors.Errorf("show terraform plan file: %w", err)
+	}
+
+	rawGraph, err := e.graph(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("graph: %w", err)
+	}
+	return ConvertResources(plan.PlannedValues.RootModule, rawGraph)
+}
+
+func (e executor) showPlan(ctx context.Context, planfilePath string) (*tfjson.Plan, error) {
+	args := []string{"show", "-json", "-no-color", planfilePath}
+	p := new(tfjson.Plan)
+	err := e.execParseJSON(ctx, args, e.basicEnv(), p)
+	return p, err
+}
+
+func (e executor) graph(ctx context.Context) (string, error) {
+	// #nosec
+	cmd := exec.CommandContext(ctx, e.binaryPath, "graph")
+	cmd.Dir = e.workdir
+	cmd.Env = e.basicEnv()
+	out, err := cmd.Output()
+	if err != nil {
+		return "", xerrors.Errorf("graph: %w", err)
+	}
+	return string(out), nil
+}
+
+// revive:disable-next-line:flag-parameter
+func (e executor) apply(ctx context.Context, env, vars []string, logger provisionersdk.Logger, destroy bool,
+) (*proto.Provision_Response, error) {
+	args := []string{
+		"apply",
+		"-no-color",
+		"-auto-approve",
+		"-input=false",
+		"-json",
+		"-refresh=true",
+	}
+	if destroy {
+		args = append(args, "-destroy")
+	}
+	for _, variable := range vars {
+		args = append(args, "-var", variable)
+	}
+
+	writer, doneLogging := provisionLogWriter(logger)
+	defer func() { <-doneLogging }()
+
+	err := e.execWriteOutput(ctx, args, env, writer)
+	if err != nil {
+		return nil, xerrors.Errorf("terraform apply: %w", err)
+	}
+	resources, err := e.stateResources(ctx)
+	if err != nil {
+		return nil, err
+	}
+	statefilePath := filepath.Join(e.workdir, "terraform.tfstate")
+	stateContent, err := os.ReadFile(statefilePath)
+	if err != nil {
+		return nil, xerrors.Errorf("read statefile %q: %w", statefilePath, err)
+	}
+	return &proto.Provision_Response{
+		Type: &proto.Provision_Response_Complete{
+			Complete: &proto.Provision_Complete{
+				Resources: resources,
+				State:     stateContent,
+			},
+		},
+	}, nil
+}
+
+func (e executor) stateResources(ctx context.Context) ([]*proto.Resource, error) {
+	state, err := e.getState(ctx)
+	if err != nil {
+		return nil, err
+	}
+	rawGraph, err := e.graph(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("get terraform graph: %w", err)
+	}
+	var resources []*proto.Resource
+	if state.Values != nil {
+		resources, err = ConvertResources(state.Values.RootModule, rawGraph)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return resources, nil
+}
+
+func (e executor) getState(ctx context.Context) (*tfjson.State, error) {
```

Comment:
This should have the `get` prefix removed to stay consistent with other similar funcs.